File: Binary Search Trees/bst_traversal.py
def in_order_traversal(tree, array):
    if tree:
        in_order_traversal(tree.left, array)
        array.append(tree.value)
        in_order_traversal(tree.right, array)
    return array

def pre_order_traversal(tree, array):
    if tree:
        array.append(tree.value)
        pre_order_traversal(tree.left, array)
        pre_order_traversal(tree.right, array)
    return array

def post_order_traversal(tree, array):
    if tree:
        post_order_traversal(tree.left, array)
        post_order_traversal(tree.right, array)
        array.append(tree.value)
    return array

File: Binary Search Trees/bstTraversal.js
const inOrderTraversal = (tree, array) => {
    if (tree) {
        inOrderTraversal(tree.left, array)
        array.push(tree.value)
        inOrderTraversal(tree.right, array)
    }
    return array
}

const preOrderTraversal = (tree, array) => {
    if (tree) {
        array.push(tree.value)
        preOrderTraversal(tree.left, array)
        preOrderTraversal(tree.right, array)
    }
    return array
}

const postOrderTraversal = (tree, array) => {
    if (tree) {
        postOrderTraversal(tree.left, array)
        postOrderTraversal(tree.right, array)
        array.push(tree.value)
    }
    return array
}